human_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -249,6 +321,144 @@
+
+
+# In[24]:
+
+
+# Load pathway data
+hallmark_DB_file = os.path.join(
+    local_dir,
+    "hallmark_DB.gmt")
+
+
+# In[25]:
+
+
+get_ipython().run_cell_magic('R', '-i template_DE_stats_file -i hallmark_DB_file -i statistic', '# This cell is temporary and contains the same code as the GSEA_analysis.R file\n# This is being used to debug the GSEA results found\n# Read in data\nDE_stats_data <- read.table(template_DE_stats_file, sep="\\t", header=TRUE, row.names=NULL)\n\n# feature 1: numeric vector\nif (statistic == \'logFC\'){\n    col_num = 2\n} else if (statistic == \'log2FoldChange\'){\n    col_num = 3\n} else if (statistic ==\'t\'){\n    col_num = 4\n} else if (statistic == \'p-value\'){\n    col_num = 5\n} else if (statistic == \'adj p-value\' || statistic == \'pvalue\'){\n    col_num = 6\n} else if ( statistic == \'padj\'){\n    col_num = 7\n}\n\nrank_genes <- as.numeric(as.character(DE_stats_data[,col_num]))\n\n# feature 2: named vector of gene ids\nnames(rank_genes) <- as.character(DE_stats_data[,1])\nprint(head(rank_genes))\n## feature 3: decreasing order\nrank_genes <- sort(rank_genes, decreasing = TRUE)\nprint(head(rank_genes))\n\npathway_DB_data <- gmtPathways(hallmark_DB_file)\nprint(head(pathway_DB_data))\n\n#print(head(pathway_DB_data))\n# GSEA is a generic gene set enrichment function\n# Different backend methods can be applied depending on the \n# type of annotations\n# Here we will use fgsea\n\n#enrich_pathways <- fgsea(pathways=pathway_DB_data,\n#                         stats=rank_genes,\n#                         nperm=10000,\n#                         minSize=10,\n#                         maxSize=200\n#                        )\n\n#print(enrich_pathways)\n\n# Look at those pathways WITH significant padj\n#plotEnrichment(pathway_DB_data[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]], stats=rank_genes, gseaParam = 1, ticksSize = 0.2)\n#plotEnrichment(pathway_DB_data[["HALLMARK_TNFA_SIGNALING_VIA_NFKB"]], stats=rank_genes, gseaParam = 1, ticksSize = 0.2)\n\n# Look at those pathways with NOT significant padj\nplotEnrichment(pathway_DB_data[["HALLMARK_P53_PATHWAY"]], stats=rank_genes, gseaParam = 1, ticksSize = 0.2)\n#barplot(sort(rank_genes, decreasing = T))"""')
+
+
+# In[26]:
+
+
+get_ipython().run_cell_magic('R', '-i template_DE_stats_file -i hallmark_DB_file -i statistic -o template_enriched_pathways', "\nsource('../generic_expression_patterns_modules/GSEA_analysis.R')\ntemplate_enriched_pathways <- find_enriched_pathways(template_DE_stats_file, hallmark_DB_file, statistic)")
+
+
+# ## Quick validation
+# Significant pathways found include: HALLMARK_G2M_CHECKPOINT, TNFA_SIGNALING, INFLAMMATORY_RESPONSE, APOPTOSIS, HYPOXIA are expected and consistent with what the publication found. However, there are pathways not found to be significant that we would expect to find: P53, NOTCH_SIGNALING, DNA_REPAIR. Not sure how to explain this since the enrichment score plot looks consistent with the p-value score .
```

Comment:
From my (very limited) cancer pathways knowledge, I'm a bit surprised that you found apoptosis as significant but not p53 signaling. I guess there are other pathways that can lead to dysregulated apoptosis in cancer, but p53 seems to be the main one, or at least the most frequently mutated one in general.

Reply:
I think this is a good point and something to check